src/matcha_ml/core/core.py

```diff
@@ -90,3 +97,80 @@ def remove_state_lock() -> None:
     """Unlock remote state."""
     remote_state = RemoteStateManager()
     remote_state.unlock()
+
+
+@track(event_name=AnalyticsEvent.PROVISION)
+def provision(
+    location: str,
+    prefix: str,
+    password: str,
+    verbose: Optional[bool] = False,
+) -> MatchaState:
+    """Provision cloud resources using templates.
+
+    Args:
+        location (str): Azure location in which all resources will be provisioned.
+        prefix (str): Prefix used for all resources.
+        password (str): Password for ZenServer.
+        verbose (bool optional): additional output is show when True. Defaults to False.
+
+    Returns:
+        MatchaState: the information of the provisioned resources.
+
+    Raises:
+        MatchaError: If resources are already provisioned.
+        MatchaError: If prefix is not valid.
+        MatchaError: If region is not valid.
+    """
+    remote_state_manager = RemoteStateManager()
+
+    if remote_state_manager.is_state_stale():
+        print_status(
```

Comment:
Is there a better way we could handle this without IO? The IO could be left in and is only shown when `verbose=True`, but that wouldn't answer the question.

One solution could be to introduce a parameter for the users to specify if they want state stale to be handled. If this is false, then an error is raised. If it's true, then `remote_matcha_config()` is called (and if `verbose=true`, the warning status is printed).

Reply:
I think that we can remove the message entirely, for the "happy" path when one user runs `matcha destroy` it should also remove all instances of the `matcha.config.json` remote state config. As this is not possible, the next best thing to do this remove the file when possible.